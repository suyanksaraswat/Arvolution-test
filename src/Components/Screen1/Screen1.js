import { Button, Input, Form, message, Row, Col, Layout } from 'antd';
import React from 'react';
import { useHistory } from 'react-router-dom';

function Screen1(props) {
    const history = useHistory();

    const onFinish = (values) => {
        history.push('/game', { player1Name: values.player1Name, player2Name: values.player2Name });
    }

    const onFinishFailed = errorInfo => {
        message.error('Enter correct name');
    };

    return (
        <Layout>
            <Layout.Content>
                <Row className='full-screen-height'>
                    <Col xs={3} sm={3} md={9} />
                    <Col xs={18} sm={18} md={6} className='column-center'>
                        <div className='login-container'>
                            <div className='text-center mb-50'>
                                <h2>Arvolution Game</h2>
                            </div>
                            <Form
                                name='basic'
                                onFinish={onFinish}
                                onFinishFailed={onFinishFailed}
                            >
                                <Form.Item
                                    name="player1Name"
                                    rules={[{ required: true, message: 'Please input player 1 name!' }]}
                                >
                                    <Input placeholder="Player 1 Name" />
                                </Form.Item>
                                <Form.Item
                                    name="player2Name"
                                    rules={[{ required: true, message: 'Please input player 2 name!' }]}
                                >
                                    <Input placeholder="Player 2 Name" />
                                </Form.Item>
                                <Form.Item className='text-center'>
                                    <Button type="primary" htmlType="submit" className='w-100'>
                                        Continue
                                    </Button>
                                </Form.Item>
                            </Form>
                        </div>
                    </Col>
                    <Col xs={3} sm={3} md={9} />
                </Row>
            </Layout.Content>
        </Layout>
    );
}
    
export default Screen1;