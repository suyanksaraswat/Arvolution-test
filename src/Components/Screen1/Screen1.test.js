import React from 'react';
import ReactDOM from 'react-dom';
import Screen1 from './Screen1';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import renderer from "react-test-renderer";

afterEach(cleanup);

it("screen1 renders without crashing", () => {
	const div = document.createElement("div");
	ReactDOM.render(<Screen1></Screen1>, div)
})

it("matches snapshot 1", () => {
	const tree = renderer.create(<Screen1></Screen1>).toJSON();
	expect(tree).toMatchSnapshot();
})