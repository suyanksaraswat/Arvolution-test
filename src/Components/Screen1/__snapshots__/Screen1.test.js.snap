// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<section
  className="ant-layout"
>
  <main
    className="ant-layout-content"
  >
    <div
      className="ant-row full-screen-height"
      style={Object {}}
    >
      <div
        className="ant-col ant-col-xs-3 ant-col-sm-3 ant-col-md-9"
        style={Object {}}
      />
      <div
        className="ant-col column-center ant-col-xs-18 ant-col-sm-18 ant-col-md-6"
        style={Object {}}
      >
        <div
          className="login-container"
        >
          <div
            className="text-center mb-50"
          >
            <h2>
              Arvolution Game
            </h2>
          </div>
          <form
            className="ant-form ant-form-horizontal"
            id="basic"
            onSubmit={[Function]}
          >
            <div
              className="ant-row ant-form-item"
              style={Object {}}
            >
              <div
                className="ant-col ant-form-item-control"
                style={Object {}}
              >
                <div
                  className="ant-form-item-control-input"
                >
                  <div
                    className="ant-form-item-control-input-content"
                  >
                    <input
                      className="ant-input"
                      id="basic_player1Name"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Player 1 Name"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              className="ant-row ant-form-item"
              style={Object {}}
            >
              <div
                className="ant-col ant-form-item-control"
                style={Object {}}
              >
                <div
                  className="ant-form-item-control-input"
                >
                  <div
                    className="ant-form-item-control-input-content"
                  >
                    <input
                      className="ant-input"
                      id="basic_player2Name"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Player 2 Name"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              className="ant-row ant-form-item text-center"
              style={Object {}}
            >
              <div
                className="ant-col ant-form-item-control"
                style={Object {}}
              >
                <div
                  className="ant-form-item-control-input"
                >
                  <div
                    className="ant-form-item-control-input-content"
                  >
                    <button
                      className="ant-btn ant-btn-primary w-100"
                      onClick={[Function]}
                      type="submit"
                    >
                      <span>
                        Continue
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
      <div
        className="ant-col ant-col-xs-3 ant-col-sm-3 ant-col-md-9"
        style={Object {}}
      />
    </div>
  </main>
</section>
`;
